api/services/itemService.js

```diff
@@ -0,0 +1,21 @@
+const { itemDao } = require("../models");
+const { validateNumber } = require("../utils/validator");
+
+const getItemDetailsById = async (itemId) => {
+  await validateNumber(itemId);
+
+  const result = await itemDao.checkIfItemExistsById(itemId);
+
+  if (!result) {
+    const error = new Error(`ITEM_DOESN'T_EXIST`);
+    error.statusCode = 404;
+
+    throw error;
+  }
+
+  return await itemDao.getItemDetailsById(itemId);
+};
```

Comment:
```suggestion
const getItemDetailsById = async (itemId) => {
  return itemDao.getItemDetailsById(itemId);
};
```
1. `validateNumber()` 함수의 정규표현식을 보면 함수에서 받는 argument가 단순 숫자인지 여부만 확인 하는 것 같습니다. 이는 controller에서 `+`를 통해 숫자로 바꾸고 있으며, 혹시 다른 문자열이 들어오더라도 itemId는 `NaN`가 되기때문에 핸들링 될 수 있을 것 같고 따로 validate 함수를 짜서 검증하는 것은 다소 verbose 하다고 보여집니다.
2. `checkIfItemExistsById()`를 통해 분기되고 있는 로직 또한 front end에서는 해당 엔드포인트에서 `404`, `ITEM_DOSEN'T_EXIST` 라는 response를 통해 핸들링을 하게 될 것 입니다. 이 경우에도 백엔드 api에서 현재 코드상 해당 itemId를 통해 query할 수 있는 결과값이 없다면 빈 배열을 반환할텐데, `ITEM_DOSEN'T_EXIST`를 통해 처리하나, 빈 배열을 통해 처리하나 똑같으나 API 코드 상에서는 로직이 이전보다 더욱 간단해지게 됩니다. 이 부분도 고려해보시면 좋을 것 같습니다.
3. 마지막 Return문에는 await가 굳이 필요없습니다.